### ISSUE TITLE
Move the SharpFont.HarfBuzz code into this repo

### ISSUE THREAD

**Issue by corliss** [Robmaister/SharpFont#103]
It would be convenient to have a single repo for all SharpFont code, including SharpFont.HarfBuzz and any other assemblies created going forward. I can't think of a good reason to have a separate repo for SharpFont.HarfBuzz, since you own both projects and since they cater to basically the same audience.

Note: The generated assemblies and nugets don't need to change - there would still be a `SharpFont` nuget and a `SharpFont.HarfBuzz` nuget. This is just about reducing the number of repos.

**Comment by HinTak** [comment]
Freetype can be built without harfbuzz. In fact I think that's the case for the windows freetype dll, since building harfbuzz on windows with Ms tools is a pain.

**Comment by corliss** [comment]
@HinTak - This change has no impact on the build. Nothing changes, except that there will be one less repo to deal with.

**Comment by HinTak** [comment]
Not really. It would be frustrating / surprising / upsetting to the end user If he/she uses a combined assembly against plain non-harfbuzz enabled freetype. If you combine the repo, the expectation would be to stop supporting non-harfbuzz-linked freetype. Not every apps needs harfbuzz - though I bundle harfbuzz-enabled freetype, strictly speaking font validator does not need any harfbuzz related functionality.

**Comment by corliss** [comment]
No, please understand, that is not what I am proposing. There will be *no change* to the assemblies or the nugets. Please re-read the original post.

**Comment by Robmaister** [comment]
SharpFont.HarfBuzz depends on SharpFont for the hafbuzz APIs that depend on freetype. I was originally thinking it would be cleaner where 1 nuget package/assembly = 1 repo, but this gets cumbersome when the assemblies have dependencies on each other.

Moving to a system where there will only ever be 2 repos (SharpFont for code, SharpFont.Dependencies for binary deps) makes sense to me. I will first merge in your PR, we can work on changing this structure, then I will do another release on NuGet (which is long overdue).

**Comment by corliss** [comment]
Great. As a first step, the .nuspec and .props files should be moved alongside their respective .csproj files (which I believe is the normal convention anyhow.)

**Comment by HinTak** [comment]
You could do it via git submodule, without merging the two repos.

**Comment by Robmaister** [comment]
This is already done via submodule: https://github.com/Robmaister/SharpFont.HarfBuzz/tree/master/Dependencies

It only serves to make the process more complicated for devs. I don't see any inherent benefits, only drawbacks of having at least two copies of the SharpFont + SharpFont.Dependencies repos, and possible submodule breakage (or forgetting to init them upon cloning).

A submodule for the binary deps makes sense as it offloads all of the big files elsewhere and keeps the main repo clean/light.
